numpy/ma/core.pyi

```diff
@@ -550,7 +552,33 @@ class _extrema_operation(_MaskedUFunc):
     def reduce(self, target, axis=...): ...
     def outer(self, a, b): ...
 
-def min(obj, axis=..., out=..., fill_value=..., keepdims=...): ...
+@overload
+def min(
+    obj: _ArrayLike[_SCT],
+    axis: None = ...,
+    out: None = ...,
+    fill_value: _ScalarLike_co | None = ...,
+    keepdims: Literal[False] = ...,
+) -> _SCT: ...
+
+@overload
+def min(
+    obj: _ArrayLike[Any],
+    axis: _ShapeLike | None = ...,
+    out: None = ...,
+    fill_value: _ScalarLike_co | None = ...,
+    keepdims: bool = ...,
+) -> Any: ...
+
+@overload
+def min(
+    obj: _ArrayLike[Any],
+    axis: _ShapeLike | None = ...,
+    out: _ArrayType = ...,
+    fill_value: _ScalarLike_co | None = ...,
+    keepdims: bool = ...,
```

Comment:
The `...` indicate that `out` is not a required argument here. But the default it is `None`, and `None` cannot be assigned to `_ArrayType`. 

I believe that pyright would report this as a warning in strict mode or something, i.e. it'll spawn one of those evil squiggly lines (sorry for the mental image, but it had to be said; hopefully you'll still be able to sleep tonight).

And in case you haven't seen this trick before, here's a cheeky little hint: 
Keep in mind that `out` can be passed positionally or by keyword.

Reply:
For the grand reveal; here's my explanation of the magic trick:
https://github.com/scipy/scipy-stubs/pull/153#discussion_r1826597984

It's not exactly the same situation, but the underlying idea is the same.

Reply:
seems like you got it right 👌🏻